packages/client/lib/CredentialOffer/CredentialOfferIssuance.ts

```diff
@@ -0,0 +1,44 @@
+import {CredentialOffer, CredentialOfferWithBaseURL, EndpointMetadata, OIDCVCIVersion} from "@sphereon/openid4vci-common";
+
+import {CredentialOfferStrategy} from "./CredentialOfferStrategy";
+
+export class CredentialOfferIssuance implements CredentialOfferStrategy {
+  readonly version: OIDCVCIVersion;
+  private readonly _credentialOfferWithBaseURL: CredentialOfferWithBaseURL;
+
+  public constructor(credentialOfferURI: string) {
+    this.version = OIDCVCIVersion.VER_11;
+    this._credentialOfferWithBaseURL = this.fromURI(credentialOfferURI);
+  }
+
+
+  public getCredentialOffer(credentialOfferURI: string): CredentialOffer {
+    return this.fromURI(credentialOfferURI);
+  }
+
+  public fromURI(credentialOfferURI: string): CredentialOfferWithBaseURL {
```

Comment:
I would expect this to be a static with a private constructor instead of doing it the other way around

Reply:
function has been converted to static.